src/common/utils/__tests__/iter.test.ts

```diff
@@ -15,4 +15,28 @@ describe("iter", () => {
       expect(reduce([], (acc: number[], current: number) => [acc[0] + current], [])).toEqual([]);
     });
   });
+
+  describe("join", () => {
```

Comment:
Sneaky.

We do not need generators for most stuff we are doing. If we did, we can do it better by using a library, or implementing something here-and-there founded on CS of functional programming. Like monads and stuff. The requirements for this sets the bar pretty high, but we really don't currently need most of that stuff either, and we do have better things to learn.

Reply:
If you want, we can experiment this very gratifying topic in another codebase?

We can study how this scales to reactive functional programming with async generators, functional generator composition, stream composition, transducers, sticky async, bells, whistles, and belgian waffles :)

Reply:
We don't need all of those features immediately.... I would much rather have an in-tree package like this to replace all of `lodash`

Reply:
1. Why reinvent `lodash` when it is already pretty useful? :)
2. Are you proposing replacing `lodash` with a pattern which is not compatible to be used freely with any function, only the internal ones? I.e. collection chaining pattern vs. functional composition.
3. What about the other stuff written previously? Writing careful responses is important, even more so with the time-difference within the team.

Reply:
1. I don't plan on reinventing `lodash`, and `lodash` itself already recommends against using it in new projects. I am merely adding things that we can use as we go to the utils folders. That way we have complete control over what it does and only have it do what we need it to do and no more.
2. See above

Reply:
> lodash itself already recommends against using it in new projects

This is news to me. I wonder why they say this. Lodash doesn't make noise about it in their website https://lodash.com.

Let's take this discussion elsewhere :)

Resolved.